components/announcement_bar/notify_admin_downgrade_delinquency_bar/index.tsx

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+
+import {FormattedMessage} from 'react-intl';
+import {useSelector, useDispatch} from 'react-redux';
+
+import {t} from 'utils/i18n';
+import {
+    AnnouncementBarTypes, MappingCloudSelfHotsSkus, PaidFeatures, Preferences, TELEMETRY_CATEGORIES,
+} from 'utils/constants';
+
+import {GlobalState} from 'types/store';
+import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
+import useGetSubscription from 'components/common/hooks/useGetSubscription';
+import {isSystemAdmin} from 'mattermost-redux/utils/user_utils';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {trackEvent} from 'actions/telemetry_actions';
+import {getSubscriptionProduct} from 'mattermost-redux/selectors/entities/cloud';
+import {NotifyStatus, useGetNotifyAdmin} from 'components/common/hooks/useGetNotifyAdmin';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+
+import AnnouncementBar from '../default_announcement_bar';
+
+export const BannerPreferenceName = 'notify_upgrade_workspace_banner';
+
+const NotifyAdminDowngradeDelinquencyBar = () => {
+    const dispatch = useDispatch();
+    const getCategory = makeGetCategory();
+    const preferences = useSelector((state: GlobalState) =>
+        getCategory(state, Preferences.NOTIFY_ADMIN_UPGRADE_DOWNGRADE_WORKSPACE),
+    );
+    const subscription = useGetSubscription();
+    const product = useSelector(getSubscriptionProduct);
+    const currentUser = useSelector((state: GlobalState) =>
+        getCurrentUser(state),
+    );
+
+    const {notifyAdmin, notifyStatus} = useGetNotifyAdmin({});
+
+    useEffect(() => {
+        if (notifyStatus === NotifyStatus.Success) {
+            dispatch(savePreferences(currentUser.id, [{
+                category: Preferences.NOTIFY_ADMIN_UPGRADE_DOWNGRADE_WORKSPACE,
+                name: BannerPreferenceName,
+                user_id: currentUser.id,
+                value: 'adminNotified',
+            }]));
+        }
+    }, [currentUser, dispatch, notifyStatus]);
+
+    const shouldShowBanner = () => {
```

Comment:
I have just extracted as a hook, that return 2 functions. One that only check the attribute and the second one, based on the first one, to check that if It have been more than 90 days since the start of delinquency state. I did this separation as the announcement bar only check if subscription isn't null and has the delinquency attribute https://github.com/mattermost/mattermost-webapp/pull/11119/files#diff-b5bd9e0bdc219d356b68771182341f56a3f93540dfd3b2d4e6ebb191b4447300